app/assets/javascripts/provider/admin/apiconfig/services/service_discovery.js

```diff
@@ -0,0 +1,107 @@
+document.addEventListener("DOMContentLoaded", (e) => {
+
+  const form_source = document.getElementById('new_service_source');
+  const form_scratch = document.querySelectorAll('form#new_service, form#create_service')[0];
```

Comment:
Just use `getElementById`?

Reply:
I think it's because there are 2 possible names for the form.

Reply:
OK got it now!